README.md

<a name="readme-top"></a>
<br />
<div align="center">
  <a href="https://github.com/othneildrew/Best-README-Template">
    <img src="media/logo.gif" alt="Logo" width="80" height="80">
  </a>

  <h3 align="center">LlamaOS</h3>

  <p align="center">
    A purely static web-based operating system designed to combat internet censorship.
    <br />
    <a href="https://github.com/GikitSRC/DuckOS/wiki"><strong><< Documentation >></strong></a>
    <br />
    <br />
    <a href="https://LlamaOS.xyz/directories/duckos.html">Quick Demo</a>
    ·
    <a href="https://main--stately-zabaione-6d17fe.netlify.app/">Unofficial Website</a>
    ·
    <a href="google.com">Google</a>
  </p>
</div>

---

<!-- ABOUT THE PROJECT -->
## About DuckOS 🪐

![Screenshot of DuckOS](media/showcase.png)

There are so many web-based operating systems out there for this purpose... (I won't call out names) but I never tought they feel like a real Operating System. That's when I got the idea of creating my own web-based operating system, with a unique look and packed with features, having a goal of being at least 90% bug-free and easy to use. But why should you choose to use DuckOS over the others?

Here's why:
* Meets the standards. Modern, simple and user-friendly interface.
* Powerful. Equipped with the best flagship proxies to help you go on unblocked sites.
* Flexible. Easily customize, develop and integrate.

Of course, DuckOS isn't created to replace your real Operating System (MacOS, Windows, etc). It will always be a project I made just for fun, with NO GOALS OF MAKING SCHOOL DISTRICTS CHASE AFTER ME

Find our website's link on the `top of the page`

<p align="right">(<a href="#readme-top">back to top</a>)</p>

---

<!-- GETTING STARTED -->
## Self-Hosting 💾
These are instructions on hosting DuckOS by yourself.

### Prerequisites

* 28MB of Disk Space (1MB if you replace the images with FontAwesome)
* Your own Domain
* Static Web Hosting Service

### Installation


1. Make a fork DuckOS's Github Repo
2. You can host DuckOS on a static web hosting service like [Netlify](https://www.netlify.com/) or [Cloudflare Pages](https://pages.cloudflare.com/) (There are many other options available) <br>

**That's it! With just two steps, you can get DuckOS up and running ;) However, if you want dynamic proxies to work, you'll have to do a bit more work and add a subdomain accordingly with the proxies app.**


<p align="right">(<a href="#readme-top">back to top</a>)</p>


---

<!-- CONTRIBUTING -->
## Contributing 🦮

Contributions are what make the open source community such an amazing place to learn, inspire, and create. Any contributions you make are **greatly appreciated**.

If you have a suggestion that would make this better, please fork the repo and create a pull request. You can also simply open an issue with the tag "enhancement".
Don't forget to give the project a star! Thanks again!

1. Fork the Project
2. Create your Feature Branch (`git checkout -b feature/AmazingFeature`)
3. Commit your Changes (`git commit -m 'Add some AmazingFeature'`)
4. Push to the Branch (`git push origin feature/AmazingFeature`)
5. Open a Pull Request

<p align="right">(<a href="#readme-top">back to top</a>)</p>

---

<!-- LICENSE -->
## License 📝

Distributed under the Apache 2.0 License. See `LICENSE.txt` for more information.

<p align="right">(<a href="#readme-top">back to top</a>)</p>

---

<!-- CONTACT -->
## Contact 📱

Arthur Jiang - [Discord Server](https://discord.gg/MxRvwQjrAa) - [Profile Page](https:/arthur-can.design/) - admin@LlamaOS.lol

Project Link: [DuckOS Demo](https://LlamaOS.xyz/directories/duckos)

_Short section right? I'm just one person working on DuckOS 😞_

<p align="right">(<a href="#readme-top">back to top</a>)</p>

---

<!-- ACKNOWLEDGMENTS -->
## Acknowledgments 🤭

Resources that made DuckOS possible.

* [Apache 2.0 License](license.txt)
* [Fontawesome](https://fontawesome.com/)
* [JQuery](https://jquery.com/)
* [Visual Studio Code](https://code.visualstudio.com/)
* [Canva](https://www.canva.com/)
* [Github](https://github.com)
  <br>
**My amazing and encouraging friends that prevented me from giving up on this project 💖**

<p align="right">(<a href="#readme-top">back to top</a>)</p>

---

<!-- DONATIONS -->
## Donations 💸

~~like~~, no, **love** the project? Consider donating to help me with the development of DuckOS!

Zelle Phone Number
```bash
+19255770178
```

Cashapp Tag
```bash
$chainedtears
```

Wise Transfer Email
```bash
1mp0ss1bl321@gmail.com
```

Ko-Fi Support Link

[![ko-fi](https://ko-fi.com/img/githubbutton_sm.svg)](https://ko-fi.com/X8X5JDVA1)



<p align="right">(<a href="#readme-top">back to top</a>)</p>





changelog.txt
==========================================================

  ________    ____   _____  
 /  _____/   /_   | /  |  | 
/   __  \     |   |/   |  |_
\  |__\  \    |   /    ^   /
 \_____  / /\ |___\____   | 
       \/  \/          |__| 
       
- Updated light mode excluding & including divs 
- Made clockwork off by default
- Background image preview in settings smaller
- Changed Welcome window text
- Removed Pokemon Black (NDS) and added more juegos to juegos app
- Window content (.window-body) border radius only apply on bottom corners
- Changed border radius of the main window to non when fullscreen is active

==========================================================

  ________    ____ .________
 /  _____/   /_   ||   ____/
/   __  \     |   ||____  \ 
\  |__\  \    |   |/       \
 \_____  / /\ |___/______  /
       \/  \/            \/ 

  TO DO:
  ------------------------------
- Update theme changing logic
- Add Pokemon Mega juegos (Maybe)
- Fix minimizing window bug
- Other minor updates & improvements
  ------------------------------
  DONE:
  ------------------------------
- Proxy warning text being aligned improperly
  ------------------------------

==========================================================

lib/scripts/window.js
let zIndexCounter = 1;
let isResizing = false;
let originalPosition;
const openedWindows = {};

function createWindow(title, content, width = '30%', height = '50%') {
  if (openedWindows[title]) {
    if (openedWindows[title].classList.contains('minimized')) {
      openedWindows[title].classList.remove('minimized');
      openedWindows[title].style.display = 'block'; 
    } else {
      openedWindows[title].style.zIndex = zIndexCounter++;
      openedWindows[title].classList.add('active');
    }
    return;
  }
  
  const window = document.createElement('div');
  window.className = 'window';
  window.style.zIndex = zIndexCounter++;
  window.style.width = width;
  window.style.height = height;
  openedWindows[title] = window;

  const maxWidth = parseFloat(width);
  const maxHeight = parseFloat(height);

  const remainingWidth = 100 - maxWidth;
  const remainingHeight = 100 - maxHeight;

  const calculatedLeft = remainingWidth / 2;
  const calculatedTop = remainingHeight / 2;

  window.style.left = calculatedLeft + '%';
  window.style.top = calculatedTop + '%';

  let originalLeft = calculatedLeft;
  let originalTop = calculatedTop;
  let originalWidth = width;
  let originalHeight = height;
  let originalzIndex;

  const header = document.createElement('div');
  header.className = 'window-header';
  header.textContent = title;

  const closeButton = document.createElement('button');
  closeButton.className = 'window-close-button';
  closeButton.innerHTML = '<abbr title="Close Window"></abbr>';
  closeButton.addEventListener('click', () => {
    window.classList.remove('active');
    setTimeout(() => {
      window.remove();
      delete openedWindows[title];
    }, 300);
  });

  const minimizeButton = document.createElement('button');
  minimizeButton.className = 'window-minimize-button';
  minimizeButton.textContent = '';
  minimizeButton.addEventListener('click', () => {
    if (!window.classList.contains('minimized')) {
      window.classList.add('minimized');
      window.style.display = 'none';
    } else {
      window.classList.remove('minimized');
      window.style.display = 'block';
      window.style.zIndex = zIndexCounter++;
      window.style.transform = 'scale(1)';
    }
  });

  const maximizeButton = document.createElement('button');
  maximizeButton.className = 'window-maximize-button';
  maximizeButton.textContent = '';
  maximizeButton.addEventListener('click', () => {
    if (!window.classList.contains('maximized')) {
      window.classList.add('maximized');
      originalLeft = window.style.left;
      originalTop = window.style.top;
      originalWidth = window.style.width;
      originalHeight = window.style.height;
      originalzIndex = window.style.zIndex;

      window.style.left = '0';
      window.style.top = '0';
      window.style.width = '100%';
      window.style.height = '100%';
      window.style.transform = 'none';
      window.style.zIndex = '101';
    } else {
      window.classList.remove('maximized');
      window.style.left = originalLeft;
      window.style.top = originalTop;
      window.style.width = originalWidth;
      window.style.height = originalHeight;
      window.style.zIndex = originalzIndex;
      window.style.transform = 'scale(1)';
    }
  });

  header.appendChild(closeButton);
  header.appendChild(minimizeButton);
  header.appendChild(maximizeButton);
  window.appendChild(header);

  const body = document.createElement('div');
  body.className = 'window-body';
  body.innerHTML = content;
  window.appendChild(body);

  let isDragging = false;
  let startPosX, startPosY, startMouseX, startMouseY;

  const bringToFront = () => {
    zIndexCounter += 1;
    window.style.zIndex = zIndexCounter;
  };

  const handleMouseDown = (e) => {
    if (!window.classList.contains('maximized')) {
      isDragging = true;
      startPosX = window.offsetLeft;
      startPosY = window.offsetTop;
      startMouseX = e.clientX;
      startMouseY = e.clientY;

      const iframes = window.querySelectorAll('iframe');
      iframes.forEach(iframe => {
        iframe.style.pointerEvents = 'none';
      });

      bringToFront();

      document.addEventListener('mousemove', handleMouseMove);
      document.addEventListener('mouseup', handleMouseUp);
      document.addEventListener('mouseleave', handleMouseUp);
    }
  };

  const handleContentClick = () => {
    bringToFront();
  };

  body.addEventListener('mousedown', handleContentClick);

  const handleMouseMove = (e) => {
    if (isDragging) {
      const offsetX = e.clientX - startMouseX;
      const offsetY = e.clientY - startMouseY;
      window.style.left = startPosX + offsetX + 'px';
      window.style.top = startPosY + offsetY + 'px';
    }
  };

  const handleMouseUp = () => {
    isDragging = false;
    document.removeEventListener('mousemove', handleMouseMove);
    document.removeEventListener('mouseup', handleMouseUp);
    document.removeEventListener('mouseleave', handleMouseUp);

    const iframes = window.querySelectorAll('iframe');
    iframes.forEach(iframe => {
      iframe.style.pointerEvents = 'auto';
    });
  };

  let isResizing = false;
  let startWidth, startHeight;

  const handleResizeMouseDown = (e) => {
    isResizing = true;
    startWidth = parseFloat(window.style.width);
    startHeight = parseFloat(window.style.height);
    startMouseX = e.clientX;
    startMouseY = e.clientY;

    const iframes = window.querySelectorAll('iframe');
    iframes.forEach(iframe => {
      iframe.style.pointerEvents = 'none';
    });

    bringToFront();

    document.addEventListener('mousemove', handleResizeMouseMove);
    document.addEventListener('mouseup', handleResizeMouseUp);
    document.addEventListener('mouseleave', handleResizeMouseUp);
  };

  const handleResizeMouseMove = (e) => {
    if (isResizing) {
      const offsetX = e.clientX - startMouseX;
      const offsetY = e.clientY - startMouseY;
      const newWidth = Math.max(30, startWidth + offsetX);
      const newHeight = Math.max(30, startHeight + offsetY);

      window.style.width = newWidth + 'px';
      window.style.height = newHeight + 'px';
      bringToFront();
    }
  };

  const handleResizeMouseUp = () => {
    isResizing = false;
    document.removeEventListener('mousemove', handleResizeMouseMove);
    document.removeEventListener('mouseup', handleResizeMouseUp);
    document.removeEventListener('mouseleave', handleResizeMouseUp);

    const iframes = window.querySelectorAll('iframe');
    iframes.forEach(iframe => {
      iframe.style.pointerEvents = 'auto';
    });
  };

  const resizeHandle = document.createElement('div');
  resizeHandle.className = 'resize-handle';
  resizeHandle.addEventListener('mousedown', handleResizeMouseDown);
  resizeHandle.addEventListener('mousemove', () => {
    if (isResizing) {
      bringToFront();
    }
  });
  window.appendChild(resizeHandle);

  header.addEventListener('mousedown', handleMouseDown);

  document.body.appendChild(window);

  setTimeout(() => {
    window.classList.add('active');
  }, 10);
}



const welcome = `
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Welcome to LlamaOS Beta!</title>
    <style>
        .container {
            max-width: 800px;
            margin: 0 auto;
            padding: 20px;
            text-align: center;
        }
        h2 {
            font-size: 35px;
            margin-bottom: 10px;
        }
        h3 {
            font-size: 20px;
            margin: 0 0 20px;
        }
        p {
            font-size: 15px;
        }
    </style>
</head>
<body>
    <div class="container">
        <h1 style="color: red;">This window will only show up once.</h1>
        <h2>Welcome to LlamaOS!</h2>
        <p>Embark on a New Digital Journey with Unrivaled Features<br><small>(Continuously Evolving)</small><br>Just Launched – Embrace the Excitement!<br><small>Ongoing Development to Enhance Your Experience</small></p>
        <p style="color: red;">Remember to enter Fullscreen so DuckOS formats correctly!</p>
        <p>Get Ready for a Polished Landing Page Coming Your Way</p>
        <hr>
        <p>Discover LlamaOS: A Glimpse into the Future</p>
        <p>Welcome to LlamaOS, where innovation meets accessibility. As you step into the world of DuckOS, you're entering an unparalleled web-based operating system, meticulously designed to redefine your digital experience. Although this version is still in its beta phase, the journey ahead promises a new era of computing freedom.</p>
        <p>Unleash Your Creativity with Customizable Settings</p>
        <p>DuckOS understands that each user is unique. That's why we've made customization a cornerstone of this OS. Tailor your environment to your preferences – adjust themes, layouts, and even workflows. Personalize your workspace like never before, reflecting your individuality.</p>
        <p>juegos On: Entertainment at Your Fingertips</p>
        <p>Get ready to juegos like never before. LlamaOS comes equipped with a vast array of captivating juegos suitable for all ages. Whether you're a casual juegosr or a hardcore enthusiast, our juegos library has something to offer. With intuitive controls and seamless integration, your gaming adventures are just a click away.</p>
        <p>Unrestricted Access with Built-In Proxies</p>
        <p>Say goodbye to online barriers. LlamaOS comes with the best flagship proxies such as Cobalt, Rammerhead and Immortal.</p>
    </div>
</body>
</html>

`

if (!localStorage.getItem('visitedBefore')) {
  createWindow('Welcome to DuckOS!', welcome, '50%', '50%');
  localStorage.setItem('visitedBefore', 'true');
}

// App's code

const proxy = `
<iframe src="/system/apps/proxies/index.html" frameborder="0" style="width: 100%; height: 100%; border: none;"></iframe>
`

const calculatorApp = `
<iframe id="ifr" src="/system/apps/calculator/index.html" style="width:100%; height:100%;" frameborder="none"></iframe>
`;

const juegos = `
<iframe id="ifr" src="/system/apps/juegos/index.html" style="width:100%; height:100%;" frameborder="none"></iframe>
`
const terminal = `
<iframe id="ifr" src="/system/apps/terminal/index.html" style="width:100%; height:100%;" frameborder="none"></iframe>
`

// List of apps 
const appsContainer = document.getElementById('appsContainer');
const data = [{
  id: 1,
  imageSrc: '/media/icons/proxy.png',
  text: 'Proxies',
  clickFunction: function() {
    createWindow('Proxies', proxy, '60%', '70%');
  }
},
{
  id: 2,
  imageSrc: '/media/icons/juegos.png',
  text: 'juegos',
  clickFunction: function() {
    createWindow('DuckOS juegos', juegos, '40%', '60%');
  }
},
{
  id: 3,
  imageSrc: '/media/icons/calculator.png',
  text: 'Calculator',
  clickFunction: function() {
    createWindow('Calculator App', calculatorApp, '15%', '50%');
  }
},
{
  id: 4,
  imageSrc: '/media/icons/terminal.png',
  text: 'Terminal',
  clickFunction: function() {
    createWindow('Terminal Emulator', terminal, '40%', '45%');
  }
},
];

function createApp(appData) {
  const miniDiv = document.createElement('div');
  miniDiv.classList.add('appsbox');

  miniDiv.setAttribute('data-app-id', appData.id);

  const image = document.createElement('img');
  image.src = appData.imageSrc;
  miniDiv.appendChild(image);

  miniDiv.addEventListener('click', appData.clickFunction);

  appsContainer.appendChild(miniDiv);
}

function removeAppById(appId) {
  const appDivToRemove = appsContainer.querySelector(`[data-app-id="${appId}"]`);
  if (appDivToRemove) {
    appsContainer.removeChild(appDivToRemove);
  }
}

// Add apps using the createApp function
data.forEach(item => {
  createApp(item);
});

// Example usage for devs: Remove the app with ID 2
// removeAppById(2);
// Example usage for devs: Remove latest added app
// removeApp(1);

/*
    const newAppData = {
        id: 39384,
        imageSrc: '/image/path',
        text: 'Name of App',
        clickFunction: function() {
            createWindow('App window name', 'innerHTML Content');
        }
    };
   
    createApp(newAppData);
*/

// IT IS RECOMMENDED TO USE STYLES INSIDE THE HTML TAGS, LIKE I DID ABOVE.

lib/scripts/basic.js
// Time logic
admin = 1;
function updateTime() {
  const currentTime = new Date();
  const hours = currentTime.getHours();
  const minutes = currentTime.getMinutes();
  const seconds = currentTime.getSeconds();
  const is24HourFormat = !twelveHourFormat;
  let formattedHours = hours;
  let amPm = '';
  if (twelveHourFormat) {
    formattedHours = hours % 12 || 12;
    amPm = hours >= 12 ? 'PM' : 'AM';
  }
  const timeString = `${formattedHours}:${padZero(minutes)}:${padZero(seconds)} ${amPm}`;
  document.getElementById('time').textContent = timeString;
}

function padZero(number) {
  return (number < 10 ? '0' : '') + number;
}
let twelveHourFormat = localStorage.getItem('twelveHourFormat') === 'true';

function toggleTimeFormat() {
  twelveHourFormat = !twelveHourFormat;
  localStorage.setItem('twelveHourFormat', twelveHourFormat.toString());
  updateTime();
}
updateTime();
setInterval(updateTime, 1000);

// Fullscreen shit

const fullscreenElement = document.getElementById('fullscreen');
fullscreenElement.addEventListener('click', toggleFullscreen);

function toggleFullscreen() {
  if (document.fullscreenElement) {
    if (document.exitFullscreen) {
      document.exitFullscreen();
    }
  } else {
    if (document.documentElement.requestFullscreen) {
      document.documentElement.requestFullscreen();
    }
  }
}

// settinglist basic functionalitie

document.addEventListener("DOMContentLoaded", function() {
  const toggleButton = document.getElementById("togglesettinglist");
  const settingslist = document.getElementById("settingslist");
  settingslist.style.zIndex = "9999";
  let altKeyPressed = false;

  function toggleSettingsList() {
    if (!settingslist.classList.contains("visible")) {
      settingslist.style.animation = "fadeIn 0.5s";
    }
    settingslist.classList.toggle("visible");
  }

  toggleButton.addEventListener("click", toggleSettingsList);

  document.addEventListener("keydown", function(event) {
    if (event.altKey && event.key === "s") {
      altKeyPressed = true;
      toggleSettingsList();
    }
  });

  document.addEventListener("keyup", function(event) {
    if (event.key === "Alt") {
      altKeyPressed = false;
    }
  });

  document.addEventListener("keyup", function(event) {
    if (!altKeyPressed && event.key === "s" && settingslist.classList.contains("visible")) {
      settingslist.style.animation = "fadeOut 0.5s";
      setTimeout(() => {
        settingslist.classList.remove("visible");
        settingslist.style.animation = "";
      }, 500);
    }
  });

  document.addEventListener("click", function(event) {
    if (!settingslist.contains(event.target) && !toggleButton.contains(event.target)) {
      if (settingslist.classList.contains("visible")) {
        settingslist.style.animation = "fadeOut 0.5s";
        setTimeout(() => {
          settingslist.classList.remove("visible");
          settingslist.style.animation = "";
        }, 500);
      }
    }
  });
});

window.addEventListener("load", function() {
  var loader = document.getElementById("loader");
  var content = document.getElementById("content");

  setTimeout(function() {
    loader.style.transition = "opacity 1s";
    loader.style.opacity = "0";
    content.style.display = "block";
    document.body.style.overflow = "auto";
  }, 3500);

  setTimeout(function() {
    loader.style.display = "none";
  }, 4500);
});


// Code for notifications

const notification = document.getElementById('notification');
let notificationClickFunction = null;
let hideTimeout;

function showNotification(title, message, clickFunction) {
  const notificationTitle = notification.querySelector('.notification-title');
  const notificationMessage = notification.querySelector('.notification-message');
  notificationTitle.textContent = title;
  notificationMessage.textContent = message;
  notification.style.display = 'block';
  notification.style.animation = 'slideIn 0.5s ease-in-out';

  notificationClickFunction = clickFunction;

  // Clear any previous timeout and set a new one
  clearHideTimeout();
  setHideTimeout();
}

function hideNotification(event) {
  if (event) {
    event.stopPropagation();
  }
  notification.style.animation = 'slideOut 0.5s ease-in-out';
  setTimeout(() => {
    notification.style.display = 'none';
    notification.style.animation = '';
  }, 500);
}

function executeNotificationFunction() {
  if (notificationClickFunction) {
    notificationClickFunction();
  }
}

function clearHideTimeout() {
  clearTimeout(hideTimeout);
}

function setHideTimeout() {
  hideTimeout = setTimeout(() => {
    hideNotification();
  }, 3000);
}

// showNotification("Notification Title", "Custom message!", customClickFunction);

setTimeout(function() {
  showNotification("Welcome to LlamaOS", "Feel free to explore the latest UI and all the exciting new features!");
}, 4500);

// Panic

let currentLink = localStorage.getItem("configuredLink") || "https://classroom.google.com";

function handleKeyPress(event) {
  if (event.key === "`") {
    const linkInput = document.getElementById("linkInput");
    const newLink = linkInput.value.trim();
    if (newLink !== "") {
      currentLink = newLink;
      localStorage.setItem("configuredLink", currentLink);
    }
    window.location.href = currentLink;
  }
}

const linkInput = document.getElementById("linkInput");
linkInput.addEventListener("input", function() {
  currentLink = linkInput.value.trim();
  localStorage.setItem("configuredLink", currentLink);
});

linkInput.value = currentLink;

document.addEventListener("keydown", handleKeyPress);

// DuckOS Loading screen tips

const splashTexts = [
  "You can press the backtick key (`) to quickly hide the page!",
  "The keybind \"Alt + S\" opens the settings menu :)",
  "The only way to do great work is to love what you do. - Steve Jobs",
  "In three words I can sum up everything I've learned about life: it goes on. - Robert Frost",
  "Success is not final, failure is not fatal: It is the courage to continue that counts. - Winston Churchill",
  "The only thing necessary for the triumph of evil is for good men to do nothing. - Edmund Burke",
  "Life is what happens when you're busy making other plans. - John Lennon",
  "The only limit to our realization of tomorrow will be our doubts of today. - Franklin D. Roosevelt",
  "The best way to predict the future is to create it. - Peter Drucker",
  "In the middle of every difficulty lies opportunity. - Albert Einstein",
  "The greatest glory in living lies not in never falling, but in rising every time we fall. - Nelson Mandela",
  "You miss 100% of the shots you don't take. - Wayne Gretzky",
  "To be yourself in a world that is constantly trying to make you something else is the greatest accomplishment. - Ralph Waldo Emerson",
  "The journey of a thousand miles begins with a single step. - Lao Tzu",
  "Success is stumbling from failure to failure with no loss of enthusiasm. - Winston Churchill",
  "Life is really simple, but we insist on making it complicated. - Confucius",
  "Don't count the days, make the days count. - Muhammad Ali",
  "The only person you are destined to become is the person you decide to be. - Ralph Waldo Emerson",
  "The harder I work, the luckier I get. - Samuel Goldwyn",
  "The secret of getting ahead is getting started. - Mark Twain",
  "The best revenge is massive success. - Frank Sinatra",
  "I can't change the direction of the wind, but I can adjust my sails to always reach my destination. - Jimmy Dean",
  "Do not dwell in the past, do not dream of the future, concentrate the mind on the present moment. - Buddha",
  "Your time is limited, don't waste it living someone else's life. - Steve Jobs",
  "The only thing standing between you and your goal is the story you keep telling yourself as to why you can't achieve it. - Jordan Belfort",
  "Don't watch the clock; do what it does. Keep going. - Sam Levenson",
  "I attribute my success to this: I never gave or took any excuse. - Florence Nightingale",
  "Life is 10% what happens to us and 90% how we react to it. - Charles R. Swindoll",
  "You are never too old to set another goal or to dream a new dream. - C.S. Lewis",
  "The only true wisdom is in knowing you know nothing. - Socrates",
  "Success usually comes to those who are too busy to be looking for it. - Henry David Thoreau",
  "The best time to plant a tree was 20 years ago. The second best time is now. - Chinese Proverb",
  "Don't let yesterday take up too much of today. - Will Rogers",
  "The road to success and the road to failure are almost exactly the same. - Colin R. Davis",
  "Life is short, and it is up to you to make it sweet. - Sarah Louise Delany",
];


function getRandomIndex(max) {
  return Math.floor(Math.random() * max);
}

const splashTextElement = document.getElementById("splashText");
const randomIndex = getRandomIndex(splashTexts.length);
splashTextElement.textContent = splashTexts[randomIndex];



const buttons = [
  { text: "Google Classroom", title: "Google Classroom", favicon: "/media/favicons/cloak/google.png" },
  { text: "Canvas", title: "Canvas  ||  Dashboard", favicon: "/media/favicons/cloak/canvas.png" },
  { text: "Defualt Title & Favicon", title: "LlamaOS", favicon: "/media/favicons/llamaos.png" }
];

function changeTabTitleAndFavicon(title, favicon) {
  document.title = title;
  const link = document.querySelector("link[rel*='icon']") || document.createElement('link');
  link.type = 'image/x-icon';
  link.rel = 'shortcut icon';
  link.href = favicon;
  document.getElementsByTagName('head')[0].appendChild(link);

  localStorage.setItem('tabTitle', title);
  localStorage.setItem('tabFavicon', favicon);
}

const buttonContainer = document.getElementById("buttonContainer");
buttons.forEach((buttonInfo) => {
  const button = document.createElement("button");
  button.textContent = buttonInfo.text;
  button.className = "disguisebtn";
  button.addEventListener("click", () => {
    changeTabTitleAndFavicon(buttonInfo.title, buttonInfo.favicon);
  });
  buttonContainer.appendChild(button);
});

const storedTabTitle = localStorage.getItem('tabTitle');
const storedTabFavicon = localStorage.getItem('tabFavicon');
if (storedTabTitle && storedTabFavicon) {
  changeTabTitleAndFavicon(storedTabTitle, storedTabFavicon);
}

const tabTitleInput = document.getElementById("tabTitleInput");
const faviconInput = document.getElementById("faviconInput");
const uploadFavicon = document.getElementById("uploadFavicon");
const setCustomFaviconButton = document.getElementById("setCustomFavicon");

setCustomFaviconButton.addEventListener("click", () => {
  const newTitle = tabTitleInput.value.trim();
  const newFavicon = faviconInput.value.trim();

  if (newTitle && newFavicon) {
    changeTabTitleAndFavicon(newTitle, newFavicon);
  }
});

// Clockwork

    (function() {
      var clockwork = document.querySelector('.clockwork');
      var isDragging = false;
      var offset = { x: 0, y: 0 };

      clockwork.addEventListener('mousedown', function(event) {
        isDragging = true;
        offset.x = event.clientX - clockwork.offsetLeft;
        offset.y = event.clientY - clockwork.offsetTop;
      });

      document.addEventListener('mouseup', function() {
        isDragging = false;
      });

      document.addEventListener('mousemove', function(event) {
        if (isDragging) {
          clockwork.style.left = event.clientX - offset.x + 'px';
          clockwork.style.top = event.clientY - offset.y + 'px';
        }
      });

      var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
      var date = new Date();
      var day = date.getDay();
      var formattedDate = date.toDateString();

      document.querySelector('.day').textContent = days[day];
      document.querySelector('.date').textContent = formattedDate;

      var centerButton = document.querySelector('.centerclockworkbtn');
      centerButton.addEventListener('click', function() {
        clockwork.style.top = '20%';
        clockwork.style.left = '50%';
      });

      var slider = document.querySelector('.slider');
      slider.addEventListener('input', function() {
        var value = this.value;
        clockwork.style.transform = 'translate(-50%, -50%) scale(' + (value / 100) + ')';
      });

      var colorPicker = document.querySelector('#color-picker');
      colorPicker.addEventListener('input', function() {
        var colorValue = this.value;
        var rgbValues = colorValue.split(',').map(function(item) {
          return parseInt(item.trim());
        });

        var red = rgbValues[0];
        var green = rgbValues[1];
        var blue = rgbValues[2];

        var h1Element = document.querySelector('.clockwork h1.day');
        var pElement = document.querySelector('.clockwork p.date');

        h1Element.style.color = 'rgb(' + red + ',' + green + ',' + blue + ')';
        pElement.style.color = 'rgb(' + red + ',' + green + ',' + blue + ')';
      });
    })();
    function toggleClockwork() {
      var clockworkDiv = document.querySelector(".clockwork");
      if (clockworkDiv.style.display === "none") {
        clockworkDiv.style.display = "block";
      } else {
        clockworkDiv.style.display = "none";
      }
    }

// Panic Button
      document.addEventListener("keydown", function(event) {
  if (event.key === "`") {
    const redirectURL = document.getElementById("panicInput").value;
    const finalRedirectURL = redirectURL ? redirectURL : "https://bing.com";
    window.location.href = finalRedirectURL;
  }
});


// Notification Center
document.addEventListener('DOMContentLoaded', function () {
            const notificationTrigger = document.getElementById('notificationbtn');
            const notificationCenter = document.getElementById('notification-center');

            notificationTrigger.addEventListener('click', function () {
                notificationCenter.classList.toggle('open');
                if (notificationCenter.classList.contains('open')) {
                    document.body.style.marginRight = '20%';
                } else {
                    document.body.style.marginRight = '0';
                }
            });

            document.addEventListener('click', function (event) {
                if (!notificationCenter.contains(event.target) && event.target !== notificationTrigger) {
                    notificationCenter.classList.remove('open');
                    document.body.style.marginRight = '0';
                }
            });
        });




let typedKeys = "";

document.addEventListener("keydown", function(event) {
  typedKeys += event.key;

  if (typedKeys === "842571") {
    //window.location.href = "/directories/adminsettings.html";
    admin = 2;
  } else if (typedKeys.length > 6) {
    typedKeys = "";
  }
});


